Update Pod OS description spec.os.name (#45845)
kubernetes/website: content/en/docs/concepts/workloads/pods/_index.md

{{< feature-state state="stable" for_k8s_version="v1.25" >}}

You should set the `.spec.os.name` field to either `windows` or `linux` to indicate the OS on
which you want the pod to run. These two are the only operating systems supported for now by
Kubernetes. In the future, this list may be expanded.

In Kubernetes v{{< skew currentVersion >}}, the value of `.spec.os.name` does not affect
how the {{< glossary_tooltip text="kube-scheduler" term_id="kube-scheduler" >}}
picks a Pod to run a node. In any cluster where there is more than one operating system for
running nodes, you should set the
[kubernetes.io/os](/docs/reference/labels-annotations-taints/#kubernetes-io-os)
label correctly on each node, and define pods with a `nodeSelector` based on the operating system

> label, the kube-scheduler assigns your pod to a node based on other criteria and may or may not
> succeed in picking a suitable node placement where the node OS is right for the containers in that Pod.

Reviewer: Quite a long sentence. How about:
```suggestion
label. If you don't specify an operating system selector for a Pod, the kube-scheduler assigns your pod
to a node based on other criteria, which could mean the Pod gets bound to a node where the node
operating system doesn't match the containers in that Pod. When that happens, the Pod can't start.
```

{{< feature-state state="stable" for_k8s_version="v1.25" >}}

You should set the `.spec.os.name` field to either `windows` or `linux` to indicate the OS on
which you want the pod to run. These two are the only operating systems supported for now by
Kubernetes. In the future, this list may be expanded.

In Kubernetes v{{< skew currentVersion >}}, the value of `.spec.os.name` does not affect

> how the [kube-scheduler](/docs/reference/command-line-tools-reference/kube-scheduler/)

Reviewer: Even better: keep the tooltip
```suggestion
how the {{< glossary_tooltip text="kube-scheduler" term_id="kube-scheduler" >}}
```

picks a Pod to run a node. In any cluster where there is more than one operating
system for running nodes, you should set the
[kubernetes.io/os](/docs/reference/labels-annotations-taints/#kubernetes-io-os)
label correctly on each node, and define pods with a `nodeSelector` based on the operating system

> label, the {{< glossary_tooltip text="scheduler" term_id="kube-scheduler" >}} assigns your pod to

Reviewer: You could omit the tooltip here if adding it earlier.

a node based on other criteria and may or may not succeed in picking a suitable node placement
where the node OS is right for the containers in that Pod.

> The [Pod security standards](/docs/concepts/security/pod-security-standards/) also uses this

Reviewer: The original text was better:
```suggestion
The [Pod security standards](/docs/concepts/security/pod-security-standards/) also use this
```
